spec/integration/hanami/controller/application_action/view_rendering/automatic_rendering_spec.rb

```diff
@@ -42,6 +42,50 @@ class Test < Main::View
     end
   end
 
+  it "Does not render a view automatically when #render? returns false " do
+    within_app do
+      write "slices/main/lib/actions/test.rb", <<~RUBY
+        require "hanami/action"
+
+        module Main
+          module Actions
+            class Test < Hanami::Action
+              def handle(req, res)
+                res[:favorite_number] = 123
+              end
+
+              def render?(_res)
+                false
+              end
+            end
+          end
+        end
+      RUBY
+
+      write "slices/main/lib/views/test.rb", <<~RUBY
```

Comment:
@cllns This has `lib` in the path.

Should we try to verify that the action works event without a view and template?

Reply:
I think we already test that later in the file, without `render?` coming into play at all. I can add an example with `render?` being `false` if you want, but it seems redundant to me

https://github.com/hanami/controller/blob/4dfd319e38a9c6c93b8469ed1708e13d1f2b4335/spec/integration/hanami/controller/application_action/view_rendering/automatic_rendering_spec.rb#L171-L191

Reply:
Good call about lib :+1: I just merged #354, rebased, and removed the `lib` from the path in these specs

Reply:
@cllns OK, if that is already tested, we're good.